src/main/java/uk/gov/digital/ho/hocs/workflow/api/bpmn/BusinessEventService.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.digital.ho.hocs.workflow.api.bpmn;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.workflow.BpmnService;
+import uk.gov.digital.ho.hocs.workflow.api.WorkflowService;
+import uk.gov.digital.ho.hocs.workflow.api.dto.GetStageResponse;
+import uk.gov.digital.ho.hocs.workflow.client.auditclient.AuditClient;
+import uk.gov.digital.ho.hocs.workflow.client.auditclient.dto.DataFieldUpdatedPayload;
+import uk.gov.digital.ho.hocs.workflow.client.caseworkclient.CaseworkClient;
+import uk.gov.digital.ho.hocs.workflow.client.caseworkclient.dto.GetCaseworkCaseDataResponse;
+import uk.gov.digital.ho.hocs.workflow.domain.model.forms.HocsFormField;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+
+import static uk.gov.digital.ho.hocs.workflow.api.WorkflowConstants.DATA_FIELD_UPDATED_EVENT_NAME;
+import static uk.gov.digital.ho.hocs.workflow.api.WorkflowConstants.PREVIOUS_VALUE_KEY;
+
+@Service
+@Slf4j
+public class BusinessEventService {
+    private final AuditClient auditClient;
+    private final CaseworkClient caseworkClient;
+    private final BpmnService bpmnService;
+    private final WorkflowService workflowService;
+
+    public BusinessEventService(AuditClient auditClient,
+                                CaseworkClient caseworkClient,
+                                BpmnService bpmnService,
+                                WorkflowService workflowService) {
+        this.auditClient = auditClient;
+        this.caseworkClient = caseworkClient;
+        this.bpmnService = bpmnService;
+        this.workflowService = workflowService;
+    }
+
+    public void createDataFieldUpdatedEvent(String caseUUIDString, String stageUUIDString, String fieldName, String newValue) {
+        UUID caseUUID = UUID.fromString(caseUUIDString);
+        UUID stageUUID= UUID.fromString(stageUUIDString);
+
+        String previousValueKey = String.format(PREVIOUS_VALUE_KEY, fieldName);
+        String name = String.format(DATA_FIELD_UPDATED_EVENT_NAME, fieldName);
+
+        GetCaseworkCaseDataResponse caseData = caseworkClient.getCase(caseUUID);
+        String previousValue = caseData.getData().get(previousValueKey);
+
+        if (Objects.equals(previousValue, newValue)) {
+            return;
+        }
+
+        GetStageResponse getStageResponse = workflowService.getStage(caseUUID, stageUUID);
```

Comment:
I don't think we should call getStage here. It gets the case from casework which duplicates line 49 with an additional call to casework and it also calls casework to assign the user as a side effect of getting the stage (which is a code smell on it's own).
As an alternative consider calling the FormService to get the schema. Either GetFormSchema with the form name or consider refactoring to use getFormWithCase which would allow you to get the case data at the same time so you could replace line 49 too.